### PR TITLE
Split out PathEnvironment and add ManagedEnvironment and RemoteEnvironment

### PR DESCRIPTION
**Pull request by mkenigs**
This is my understanding of what we settled on doing. One thing we didn't discuss is where `Floxmeta` will live, but I think this is an acceptable starting point

### CODE REVIEW

**Review by zmitchell — APPROVED** [review]

File: crates/flox-rust-sdk/src/models/environment/mod.rs [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+    AbortTransaction(std::io::Error),
+    #[error("Failed to remove transaction backup: {0}")]
+    RemoveBackup(std::io::Error),
+    #[error("Failed to copy file")]
+    CopyFile(IoError),
+}
+
+/// Within a nix AST, find the first definition of an attribute set,
+/// that is not part of a `let` expression or a where clause
+fn find_attrs(mut expr: Expr) -> Result<AttrSet, ()> {
```
[the remark this answers is not in the document]
Comment (mkenigs) [reply crates/flox-rust-sdk/src/models/environment/mod.rs:155]: I'm not sure but I don't think changing that belongs in this PR

File: crates/flox-rust-sdk/src/models/environment/mod.rs [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+}
+
+pub enum ManifestContent {
+    Unchanged,
+    Changed(String),
+}
+
+/// insert packages into the content of a flox.nix file
+///
+/// TODO: At some point this should return None if the contents were not changed,
```
[the remark this answers is not in the document]
Comment (mkenigs) [reply crates/flox-rust-sdk/src/models/environment/mod.rs:214]: Updating

File: crates/flox-rust-sdk/src/models/environment/mod.rs [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+    let green_tree = config_attrset
+        .syntax()
+        .replace_with(edited.syntax().green().into_owned());
+    let new_content = nixpkgs_fmt::reformat_string(&green_tree.to_string());
+    Ok(ManifestContent::Changed(new_content))
+}
+
+/// remove packages from the content of a flox.nix file
+///
+/// TODO: At some point this should indicate whether the contents were actually changed
```
[the remark this answers is not in the document]
Comment (mkenigs) [reply crates/flox-rust-sdk/src/models/environment/mod.rs:261]: Updating

File: crates/flox-rust-sdk/src/models/environment/managed_environment.rs [hunk 4]
```diff
@@ -0,0 +1,88 @@
+use async_trait::async_trait;
+use flox_types::catalog::EnvCatalog;
+use runix::command_line::NixCommandLine;
+
+use super::{Environment, EnvironmentError2};
+use crate::models::environment_ref::{EnvironmentName, EnvironmentOwner, EnvironmentRef};
+use crate::prelude::flox_package::FloxPackage;
+
+#[derive(Debug)]
+pub struct ManagedEnvironment {}
```
[the remark this answers is not in the document]
Comment (mkenigs) [reply crates/flox-rust-sdk/src/models/environment/managed_environment.rs:10]: Updating

**Comment by mkenigs** [comment]
No user facing changes

**Pull request merged**